Compilation failure with json_parsing.cpp post 6d0e2b7

Hi,
Cpprest build fails with the below errors,


src/json/json_parsing.cpp:1077:20: error: could not convert ‘value’ from ‘std::unique_ptr<web::json::details::_String, std::default_delete<web::json::details::_String> >’ to ‘std::unique_ptr<web::json::details::_Value>’
             return value;
                    ^
Release/src/json/json_parsing.cpp:1089:20: error: could not convert ‘value’ from ‘std::unique_ptr<web::json::details::_Number, std::default_delete<web::json::details::_Number> >’ to ‘std::unique_ptr<web::json::details::_Value>’
             return value;
                    ^
src/json/json_parsing.cpp:1096:20: error: could not convert ‘value’ from ‘std::unique_ptr<web::json::details::_Number, std::default_delete<web::json::details::_Number> >’ to ‘std::unique_ptr<web::json::details::_Value>’
             return value;
                    ^
Release/src/json/json_parsing.cpp:1103:20: error: could not convert ‘value’ from ‘std::unique_ptr<web::json::details::_Boolean, std::default_delete<web::json::details::_Boolean> >’ to ‘std::unique_ptr<web::json::details::_Value>’
             return value;
                    ^


Release/src/json/json_parsing.cpp:1043:28:error: could not convert ‘result’ from ‘std::unique_ptr<web::json::details::_Array, std::default_delete<web::json::details::_Array> >’ to ‘std::unique_ptr<web::json::details::_Value>’
                     return result;


I am on gcc 4.8 and boost_1_68.

This has not been observed in the earlier commit cdae258.

Am I missing something during the build?

Thanks & Regards,
Swathi Desai

--- a/Release/src/json/json_parsing.cpp
+++ b/Release/src/json/json_parsing.cpp
@@ -1040,7 +1040,7 @@ std::unique_ptr<web::json::details::_Value> JSON_Parser<CharType>::_ParseArray(
                 case JSON_Parser<CharType>::Token::TKN_CloseBracket:
                     GetNextToken(tkn);
                     if (tkn.m_error) return utility::details::make_unique<web::json::details::_Null>();
-                    return result;
+                    return std::unique_ptr<web::json::details::_Value>(result.release());
                 default:
                     SetErrorCode(tkn, json_error::malformed_array_literal);
                     return utility::details::make_unique<web::json::details::_Null>();
@@ -1058,6 +1058,7 @@ template<typename CharType>
 std::unique_ptr<web::json::details::_Value> JSON_Parser<CharType>::_ParseValue(
     typename JSON_Parser<CharType>::Token& tkn)
 {
+    typedef std::unique_ptr<web::json::details::_Value> Vptr;
     switch (tkn.kind)
     {
         case JSON_Parser<CharType>::Token::TKN_OpenBrace:
@@ -1070,7 +1071,7 @@ std::unique_ptr<web::json::details::_Value> JSON_Parser<CharType>::_ParseValue(
         }
         case JSON_Parser<CharType>::Token::TKN_StringLiteral:
         {
-            auto value = utility::details::make_unique<web::json::details::_String>(std::move(tkn.string_val),
+            Vptr value = utility::details::make_unique<web::json::details::_String>(std::move(tkn.string_val),
                                                                                     tkn.has_unescape_symbol);
             GetNextToken(tkn);
             if (tkn.m_error) return utility::details::make_unique<web::json::details::_Null>();
@@ -1078,7 +1079,7 @@ std::unique_ptr<web::json::details::_Value> JSON_Parser<CharType>::_ParseValue(
         }
         case JSON_Parser<CharType>::Token::TKN_IntegerLiteral:
         {
-            std::unique_ptr<web::json::details::_Number> value;
+            Vptr value;
             if (tkn.signed_number)
                 value = utility::details::make_unique<web::json::details::_Number>(tkn.int64_val);
             else
@@ -1090,14 +1091,14 @@ std::unique_ptr<web::json::details::_Value> JSON_Parser<CharType>::_ParseValue(
         }
         case JSON_Parser<CharType>::Token::TKN_NumberLiteral:
         {
-            auto value = utility::details::make_unique<web::json::details::_Number>(tkn.double_val);
+            Vptr value = utility::details::make_unique<web::json::details::_Number>(tkn.double_val);
             GetNextToken(tkn);
             if (tkn.m_error) return utility::details::make_unique<web::json::details::_Null>();
             return value;
         }
         case JSON_Parser<CharType>::Token::TKN_BooleanLiteral:
         {
-            auto value = utility::details::make_unique<web::json::details::_Boolean>(tkn.boolean_val);
+            Vptr value = utility::details::make_unique<web::json::details::_Boolean>(tkn.boolean_val);
             GetNextToken(tkn);
             if (tkn.m_error) return utility::details::make_unique<web::json::details::_Null>();
             return value;

